pkg/services/profiles/add.go

```diff
@@ -0,0 +1,195 @@
+		Namespace: availableProfile.HelmRepository.Namespace,
+	}
+
+	newRelease := helm.MakeHelmRelease(opts.Name, version, opts.Cluster, opts.Namespace, helmRepo)
+
+	files, err := gitProvider.GetRepoDirFiles(ctx, configRepoUrl, git.GetSystemPath(opts.Cluster), defaultBranch)
+	if err != nil {
+		return fmt.Errorf("failed to get files in '%s' for config repository '%s': %s", git.GetSystemPath(opts.Cluster), configRepoUrl.String(), err)
+	}
+
+	file, err := AppendProfileToFile(files, newRelease, git.GetProfilesPath(opts.Cluster, models.WegoProfilesPath))
+	if err != nil {
+		return fmt.Errorf("failed to append HelmRelease to profiles file: %w", err)
+	}
+
+	pr, err := gitProvider.CreatePullRequest(ctx, configRepoUrl, gitproviders.PullRequestInfo{
+		Title:         fmt.Sprintf("GitOps add %s", opts.Name),
+		Description:   fmt.Sprintf("Add manifest for %s profile", opts.Name),
+		CommitMessage: AddCommitMessage,
+		TargetBranch:  defaultBranch,
+		NewBranch:     automation.GetRandomString("wego-"),
+		Files:         []gitprovider.CommitFile{file},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create pull request: %s", err)
+	}
+
+	s.Logger.Actionf("Pull Request created: %s", pr.Get().WebURL)
+
+	if opts.AutoMerge {
+		s.Logger.Actionf("auto-merge=true; merging PR number %v", pr.Get().Number)
+
+		if err := gitProvider.MergePullRequest(ctx, configRepoUrl, pr.Get().Number, AddCommitMessage); err != nil {
+			return fmt.Errorf("error auto-merging PR: %w", err)
+		}
+	}
+
+	s.printAddSummary(opts)
+
+	return nil
+}
+
+func (s *ProfilesSvc) printAddSummary(opts AddOptions) {
+	s.Logger.Println("Adding profile:\n")
+	s.Logger.Println("Name: %s", opts.Name)
+	s.Logger.Println("Version: %s", opts.Version)
+	s.Logger.Println("Cluster: %s", opts.Cluster)
+	s.Logger.Println("Namespace: %s\n", opts.Namespace)
+}
+
+// AppendProfileToFile appends a HelmRelease to profiles.yaml if file does not contain other HelmRelease with the same name and namespace.
+func AppendProfileToFile(files []*gitprovider.CommitFile, newRelease *v2beta1.HelmRelease, path string) (gitprovider.CommitFile, error) {
+	var content string
+
+	for _, f := range files {
+		if f.Path != nil && *f.Path == path {
+			if f.Content == nil || *f.Content == "" {
+				break
+			}
+
+			manifestByteSlice, err := splitYAML([]byte(*f.Content))
+			if err != nil {
+				return gitprovider.CommitFile{}, fmt.Errorf("error splitting %s: %w", models.WegoProfilesPath, err)
+			}
+
+			for _, manifestBytes := range manifestByteSlice {
+				var r v2beta1.HelmRelease
+				if err := yaml.Unmarshal(manifestBytes, &r); err != nil {
+					return gitprovider.CommitFile{}, fmt.Errorf("error unmarshaling %s: %w", models.WegoProfilesPath, err)
+				}
+
+				if profileIsInstalled(r, *newRelease) {
+					return gitprovider.CommitFile{}, fmt.Errorf("version %s of profile '%s' already exists in namespace %s", r.Spec.Chart.Spec.Version, r.Name, r.Namespace)
+				}
+			}
+
+			content = *f.Content
```

Comment:
Yes, however, it's a good point that if we only care about a single path, we should break the loop at this point, since we found it, there is no point in letting it run over the rest of the files. :)